### PR TITLE
[mistralai]: Add langchain-mistralai partner package

### PR DESCRIPTION
**Pull request by baskaryan**

### CODE REVIEW

**Comment by baskaryan** [comment]
integration test ✅  https://github.com/langchain-ai/langchain/actions/runs/7228086634/job/19696897172

**Comment by baskaryan** [comment]
address #14763

**Comment by thehunmonkgroup** [comment]
This extension of my original work solves the remaining issue over at #14775 -- I vote we proceed with this PR instead, closing my original one.

File: libs/partners/mistralai/tests/integration_tests/test_chat_models.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+"""Test ChatMistral chat model."""
+from langchain_mistralai.chat_models import ChatMistralAI
+
+
+def test_stream() -> None:
+    """Test streaming tokens from OpenAI."""
```
Comment (TomTom101) [inline libs/partners/mistralai/tests/integration_tests/test_chat_models.py:6]: … from MistralAI

**Pull request merged**